### PR TITLE
Refactor/removal of hof controllers

### PR DESCRIPTION
**Pull request by joefitter**
Refactor - integrate hof controllers
* added base-controller from hof-controllers as default export. Extending from what was previously /lib/form.js
* added error from hof-controllers as default error
* Fixed bug in tests where form.formatters were being mutated - these can now be set on req.form.options.formatters
* added hof-behaviour-session and hof-behaviour-hooks - mixin to exported controller
* Removed lint for now as mix of es5/es6

### CODE REVIEW

File: lib/base-error.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+var _ = require('underscore');
```
Comment (lennym) [inline lib/base-error.js:1]: Probably not in scope for now, but I'd been keen to know how far we are from `s/underscore/lodash`.

File: lib/controller.js [hunk 2]
```diff
@@ -0,0 +1,148 @@
+'use strict';
+
+const _ = require('lodash');
```
Comment (lennym) [inline lib/controller.js:3]: Actually, maybe it is in scope... having `underscore` *and* `lodash` in the same project is N.A.S.T.Y.

Comment (joefitter) [reply lib/controller.js:3]: agreed, 100%, which is why my next PR will remove it

Comment (joefitter) [reply lib/controller.js:3]: just tooo big to do it all in one

File: lib/base-error.js (hunk 1 above)
Comment (joefitter) [reply lib/base-error.js:1]: this will follow, promise x

Comment (lennym) [reply lib/base-error.js:1]: I *think* the only thing that gets lost from underscore transitioning to lodash is being able to pass a scope as an argument to an iterator, which we can get around pretty easily.

**Comment by joefitter** [comment]
@lennym tell you what, why don't I PR the es6 refactor into this branch? 💡

**Comment by lennym** [comment]
I don't think there's anything fundamental that should stop this going in now, as long as the things above are addressed before we publish.

**Pull request merged**